### ISSUE TITLE
POST handler being called on an OPTIONS request

### ISSUE THREAD

**Issue by jryan128** [jooby-project/jooby#2863]
![image](https://user-images.githubusercontent.com/640226/229861724-37992bb9-ac2a-46c3-a6f6-d02f61635f8c.png)

I would expect 405 Method Not Allowed not for it to be routed to the post handler?

**Comment by jknack** [comment]
can you add an example?

I don't think your handler is executed. But I do see this:

https://github.com/jooby-project/jooby/blob/12cd98d3eb08a7f43483108c5783b4b0a7633368/jooby/src/main/java/io/jooby/Route.java#L327-L328

@imeszaros do you remember why we send `200` for options?

**Comment by jknack** [comment]
Here is a test that ensure the handler is not executed:

```java
public class Issue2863 {

  @ServerTest
  public void shouldBeAbleToSendMessageOnConnect(ServerTestRunner runner) {
    runner
        .define(app -> {
          app.post("/2863", ctx -> ctx.getRequestPath());
        })
        .ready(
            client -> {
              client
                  .options("/2863",
                      rsp -> {
                        assertEquals("", rsp.body().string());
                        assertEquals(200, rsp.code());
                      });
            });
  }
}
``` 

We just get the `200` response as pointed out previously

**Comment by jryan128** [comment]
[Exact code](https://gist.github.com/jryan128/446acc6ace9d67200db37031ec3c6517)

Image of debugger paused inside `put` handler even though it came in as `options`
![image](https://user-images.githubusercontent.com/640226/230626399-1b9d29e2-361f-4e51-8ed0-f59b03010fc0.png)
```
ctx = {NettyContext@17841} "OPTIONS /"
 streamId = null
 setHeaders = {DefaultHttpHeaders@17903} "DefaultHttpHeaders[date: Fri, 7 Apr 2023 14:29:34 GMT, server: N, content-type: text/plain]"
 bufferSize = 16384
 decoder = null
 router = {App@17876} "App:0.0.0"
 route = {Route@17904} "POST /"
 ctx = {DefaultChannelHandlerContext@17905} "ChannelHandlerContext(handler, [id: 0x73682045, L:/127.0.0.1:8080 - R:/127.0.0.1:55758])"
 req = {DefaultHttpRequest@17906} "DefaultHttpRequest(decodeResult: success, version: HTTP/1.1)\r\nOPTIONS /?password=asdf HTTP/1.1\r\nHost: localhost:8080\r\nUser-Agent: insomnia/2023.1.0\r\nAccept: */*"
 path = "/"
 status = {HttpResponseStatus@17908} "200 OK"
 responseStarted = false
 query = null
 form = null
 multipart = null
 files = null
 headers = null
 pathMap = {Collections$EmptyMap@17909}  size = 0
 responseType = null
 attributes = {HashMap@17910}  size = 0
 contentLength = -1
 needsFlush = false
 cookies = null
 responseCookies = null
 resetHeadersOnError = null
 webSocket = null
 method = "OPTIONS"
 listeners = {CompletionListeners@17912} 
 remoteAddress = null
 host = null
 scheme = null
 port = 0
```
Stacktrace:
```
lambda$new$41a7b78$2:50, App (app)
apply:-1, App$$Lambda$1918/0x0000000801cdf7a0 (app)
lambda$apply$e67b40fd$1:101, CorsHandler (io.jooby)
apply:-1, CorsHandler$$Lambda$1951/0x0000000801d003d0 (io.jooby)
lambda$apply$e67b40fd$1:55, GracefulShutdownHandler (io.jooby.internal)
apply:-1, GracefulShutdownHandler$$Lambda$1952/0x0000000801d00a20 (io.jooby.internal)
lambda$then$e67b40fd$1:90, Route$Decorator (io.jooby)
apply:-1, Route$Decorator$$Lambda$1929/0x0000000801ce1780 (io.jooby)
apply:28, CompletionStageHandler (io.jooby.internal.handler)
detach:228, NettyContext (io.jooby.internal.netty)
apply:21, DetachHandler (io.jooby.internal.handler)
execute:34, StaticRouterMatch (io.jooby.internal)
channelRead:106, NettyHandler (io.jooby.internal.netty)
invokeChannelRead:379, AbstractChannelHandlerContext (io.netty.channel)
invokeChannelRead:365, AbstractChannelHandlerContext (io.netty.channel)
fireChannelRead:357, AbstractChannelHandlerContext (io.netty.channel)
fireChannelRead:436, CombinedChannelDuplexHandler$DelegatingChannelHandlerContext (io.netty.channel)
fireChannelRead:336, ByteToMessageDecoder (io.netty.handler.codec)
channelRead:308, ByteToMessageDecoder (io.netty.handler.codec)
channelRead:251, CombinedChannelDuplexHandler (io.netty.channel)
invokeChannelRead:379, AbstractChannelHandlerContext (io.netty.channel)
invokeChannelRead:365, AbstractChannelHandlerContext (io.netty.channel)
fireChannelRead:357, AbstractChannelHandlerContext (io.netty.channel)
channelRead:1410, DefaultChannelPipeline$HeadContext (io.netty.channel)
invokeChannelRead:379, AbstractChannelHandlerContext (io.netty.channel)
invokeChannelRead:365, AbstractChannelHandlerContext (io.netty.channel)
fireChannelRead:919, DefaultChannelPipeline (io.netty.channel)
read:166, AbstractNioByteChannel$NioByteUnsafe (io.netty.channel.nio)
processSelectedKey:788, NioEventLoop (io.netty.channel.nio)
processSelectedKeysOptimized:724, NioEventLoop (io.netty.channel.nio)
processSelectedKeys:650, NioEventLoop (io.netty.channel.nio)
run:553, NioEventLoop (io.netty.channel.nio)
run:997, SingleThreadEventExecutor$4 (io.netty.util.concurrent)
run:74, ThreadExecutorMap$2 (io.netty.util.internal)
run:30, FastThreadLocalRunnable (io.netty.util.concurrent)
run:1589, Thread (java.lang)
```

**Comment by jknack** [comment]
Is it possible client didn't set the `Origin` header?

I'm able to reproduce it when client doesn't set the `Origin` header.

**Comment by jryan128** [comment]
Yes. The client did not set Origin.

**Comment by jknack** [comment]
Fixed in a25c841cf520630bd34cd8079f0bbf92cc4dd955

**Comment by jryan128** [comment]
Thanks for the quick resolution. I'll do a test soon.
